File: BEmu/MarketDataRequest/ElementMarketServiceStatus.cs
//------------------------------------------------------------------------------
// <copyright project="BEmu" file="MarketDataRequest/ElementMarketServiceStatus.cs" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

namespace BEmu.MarketDataRequest
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    internal class ElementMarketServiceStatus : Element
    {
        private readonly ElementMarketString _serviceName;

        internal ElementMarketServiceStatus(MessageMarketServiceStatus arg)
        {
            this._serviceName = (ElementMarketString)arg.Elements.First();
        }

        public override int NumElements { get { return 1; } }
        public override int NumValues { get { return 1; } }
        public override Name Name { get { return new Name("ServiceOpened"); } }
        public override bool IsArray { get { return false; } }
        public override bool IsComplexType { get { return true; } }
        public override bool IsNull { get { return false; } }
        public override object this[int index] { get { return null; } }

        

        public override Element GetElement(int index)
        {
            if (index == 0)
                return this._serviceName;
            else
                return base.GetElement(index);
        }

        internal override StringBuilder PrettyPrint(int tabIndent)
        {
            StringBuilder result = new StringBuilder();
            result.AppendFormat("{0} = {{{1}", this.Name.ToString(), Environment.NewLine);
            result.Append(this._serviceName.PrettyPrint(1));
            result.Append("}" + Environment.NewLine);
            return result;
        }
    }
}

File: BEmuJava/bemu/src/main/java/com/bemu/BEmu/MarketDataRequest/ElementMarketServiceStatus.java
//------------------------------------------------------------------------------
// <copyright project="BEmuJava" file="BEmu.MarketDataRequest.ElementMarketServiceStatus.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bemu.BEmu.MarketDataRequest;

import com.bemu.BEmu.Element;
import com.bemu.BEmu.Name;

public class ElementMarketServiceStatus extends Element
{
    private final ElementMarketString _serviceName;

    ElementMarketServiceStatus(MessageMarketServiceStatus arg) throws Exception
    {
        this._serviceName = (ElementMarketString)arg.getElement("serviceName");
    }
	
    public Name name()
    {
    	return new Name("ServiceOpened");
    }
    
    public int numValues()
    {
    	return 1;
    }
    
    public int numElements()
    {
    	return 1;
    }
    
    public boolean isComplexType()
    {
    	return true;
    }
    
    public boolean isArray()
    {
    	return false;
    }
    
    public boolean isNull()
    {
    	return false;
    }
    
    protected StringBuilder prettyPrint(int tabIndent)
    {
        StringBuilder result = new StringBuilder();

        result.append(String.format("%s = {%s", this.name().toString(), System.getProperty("line.separator")));
        result.append(this._serviceName.prettyPrint(tabIndent + 1));
        result.append(String.format("}", System.getProperty("line.separator")));

        return result;
    }
    
}
